src/transformers/models/mpnet/modeling_tf_mpnet.py

```diff
@@ -1088,7 +1094,7 @@ def __init__(self, config, *inputs, **kwargs):
         super().__init__(config, *inputs, **kwargs)
         self.num_labels = config.num_labels
 
-        self.mpnet = TFMPNetMainLayer(config, name="mpnet")
+        self.mpnet = TFMPNetMainLayer(config, add_pooling_layer=False, name="mpnet")
```

Comment:
Is this backwards compatible? It looks like the pooling layer would have been added previously

Reply:
It isn't, but as-is the model is incompatible with Torch (and was causing tests to fail) because the TF models were gaining an extra layer that the Torch ones didn't have! I think this is the lesser of two evils, since I don't think there are too many MPNet saves floating around.

Reply:
> but as-is the model is incompatible with Torch

My understanding it that the model would still load from PT weights, but would randomly initialize these layers. Are there any other incompatibilities? 

> I don't think there are too many TF MPNet saves floating around.

Unfortunately, we can't know TF vs. PT downloads from the hub, but there's a handful of repos which have a decent number os downloaded e.g. [here](https://huggingface.co/sentence-transformers/paraphrase-multilingual-mpnet-base-v2/tree/main) and [here](https://huggingface.co/sentence-transformers/nli-mpnet-base-v2/tree/main). 

It's arguably making the model "correct" but I believe this will drop layers for pretained checkpoints that previously had pooling layers - effecting finetuned models and downstream tasks. 

We should try and find a way to make sure the layers are kept for older models but not added to newer ones

Reply:
Good point, actually - I'll revert these changes and skip the TF-PT compatibility check for now.

Reply:
I'm not sure it's possible to have different behaviour for old and new models, though - weights are initialized before being loaded from the checkpoint, so can't really change up the config based on whether the checkpoint has a certain tensor or not!

Still, you're definitely right that preserving backward compatibility matters more here, so I'll just revert the changes and patch the tests.